pkg/apis/pipeline/v1/matrix_types.go

```diff
@@ -96,11 +97,47 @@ func createCombination(name string, value string, combination Params) Params {
 
 // CountCombinations returns the count of combinations of Parameters generated from the Matrix in PipelineTask.
 func (m *Matrix) CountCombinations() int {
+	// Explicit combinations in the Matrix
+	if m.hasInclude() && !m.hasParams() {
+		return len(m.Include)
+	}
+
+	// Iterate over matrix.params and compute count of all generated combinations
+	count := countMatrixParamsCombinations(m.Params)
+
+	if m.hasInclude() {
+		paramMap := m.Params.extractParamMapArrVals()
+		// Add any additional combinations generated from matrix include params
+		count += countNewCombinationsFromInclude(m.Include, paramMap)
```

Comment:
no need to pass `m.Include` as a param, declare `countNewCombinationsFromInclude` as a member function `func (m *Matrix) countNewCombinationsFromInclude() {...}`